mobile-browser-based-version/server/topologies.js

```diff
@@ -92,6 +96,8 @@ class BinaryTree {
             deepestParent.rightChild = null
         }
         delete this.index[nodeId]
+
+        return affectedPeers
```

Comment:
API-wise it seems a little uninutitive to me that the `removePeer` frunction returns `affectedPeers`. If it you want this behavior, I think `addPeer` should do the same.

I think the code would be easiest to understand if the trees are immutable, and `removePeer` returns a new tree. You can then have a separate function `topologyDiff` or something to find out which peers are affected. But I can also live with this, if you prefer it.

Reply:
I agree that `addPeer` should also return `affectedPeers` since we need to notify them (in this case only the parent) as well, but I think copying the entire tree on every addition and removal is not really scalable in a dynamic system with a lot of nodes.

Reply:
In terms of scalability, there are very nice libraries for immutable data structures (https://immutable-js.github.io/immutable-js/) that don't actually need to duplicate the whole tree if you edit it (although it reliably behaves like you duplicated the whole tree). But no need to look into that now.